### ISSUE TITLE
Accessible name does not match or contain the visible label text

### ISSUE THREAD

**Issue by githubbikassahu** [ckeditor/ckeditor5#15883]
## 📝 Provide detailed reproduction steps (if any)

1. Implement a CK editor text area
2. Scan for accessibility check with IBM Accessibility
Equal Access Toolkit
3.

### ✔️ Expected result

_What is the expected result of the above steps?_
Expectation is there should be zero violation. 

### ❌ Actual result
![image](https://github.com/ckeditor/ckeditor5/assets/160510087/6b72cb78-e29c-4884-8b24-da5766ed6c41)

_What is the actual result of the above steps?_
We are getting 1 violation for the CK editor but unable to access the code. 
### ❓ Possible solution

_If you have ideas, you can list them here. Otherwise, you can delete this section._

## 📃 Other details

* Browser: …
* OS: …
* First affected CKEditor version: …
* Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Witoso** [comment]
Hmm, I've never seen this issue before. @Comandeer as you're on the topic right now, could you share your thoughts?

**Comment by Comandeer** [comment]
It seems to be a failure of the [WCAG 2.5.3 Label in name](https://www.w3.org/WAI/WCAG22/quickref/#label-in-name) success criterion. The current accessible name of the dropdown menu is set via the `aria-label` attribute to "Heading", while the visible label shows the currently selected style. It could create issue for e.g. [users navigating with voice control](https://www.sarasoueidan.com/blog/accessible-text-labels/).

We could add the currently selected style as the first part of the label and keep the name of the input as the second part, e.g. "Paragraph, Heading". This technique is [used in CKEditor 4](https://ckeditor.com/docs/ckeditor4/latest/examples/format.html).
